assembly/src/assembler/instruction/crypto_ops.rs

```diff
@@ -96,19 +96,58 @@ pub(super) fn mtree_get(span: &mut SpanBuilder) -> Result<Option<CodeBlock>, Ass
 ///
 /// This operation takes 14 VM cycles.
 pub(super) fn mtree_set(span: &mut SpanBuilder) -> Result<Option<CodeBlock>, AssemblyError> {
+    // input: [d, i, R_old, V_new, ...]
+
     // Inject the old node value onto the stack for the call to MRUPDATE.
-    // [d, i, R, V_new, ...] => [V_old, d, i, R, V_new, ...]
+    // stack: [V_old, d, i, R_old, V_new, ...]
     read_mtree_node(span);
+
+    // Note: The stack is 14 elements deep already. The maximum depth the VM can go is 16, so it's
+    // only possible to copy 2-elements at the time.
+    #[rustfmt::skip]
+    let copy_old_element_ops = [
+        // Renamed V_old => o<pos>, R_old => r<pos>, V_new => n<pos>
+        // stack: [[o0, o1, o2, o3], [d, i, r0, r1], [r2, r3, n0, n1], n2, n3, ...]
```

Comment:
Should the elements be in order `[o3, o2, o1, o0]` instead of `[o0, o1, o2, o3]`?